### PR TITLE
activate redux-immutable-state-invariant

### PR DESCRIPTION
**Pull request by sheldonwjones**
Utilize redux-immutable-state-invariant to help with development.

https://github.com/coryhouse/react-slingshot/blob/master/package.json#L79

### CODE REVIEW

File: src/store/configureStore.dev.js [hunk 1]
```diff
@@ -3,13 +3,17 @@
 // With Redux, the actual stores are in /reducers.
 
 import {createStore, compose, applyMiddleware} from 'redux';
+import reduxImmutableStaeInvariant from 'redux-immutable-state-invariant'
```
Comment (coryhouse) [inline src/store/configureStore.dev.js]: There's a typo in the variable name. :)

**Comment by sheldonwjones** [comment]
Sorry about that. I copied that line from a project that has src/.eslintrc set to.
`
{
  "rules": {
    "semi": [2, "never"]
  }
}
`

File: src/store/configureStore.dev.js [hunk 2]
```diff
@@ -3,13 +3,17 @@
 // With Redux, the actual stores are in /reducers.
 
 import {createStore, compose, applyMiddleware} from 'redux';
+import reduxImmutableStaeInvariant from 'redux-immutable-state-invariant';
```
Comment (coryhouse) [inline src/store/configureStore.dev.js]: reduxImmutableStaeInvariant is missing a "t". Should be reduxImmutableStateInvariant.

:)

**Comment by coryhouse** [comment]
@kwelch Thoughts on this? Worth accepting?

**Comment by kwelch** [comment]
I had never seen this before. I am for it, very useful for a new developer, especially if they are not use to immutability & pure functions. 

It even caught some errors that happen to me when pulling data from state.  💯

**Comment by sheldonwjones** [comment]
A better name might be immutableStateInvariantMiddleware or even reduxImmutableStateInvariantMiddleware. I'm not sure the best practice here.

Thoughts?

**Comment by coryhouse** [comment]
Thanks @sheldonwjones ! 👍

**Pull request merged**